docs/tutorials/gallery_adding_an_assistant.py

```diff
@@ -0,0 +1,110 @@
+"""
+# Adding an Assistant
+
+Ragna has builtin support for [several assistants][ragna.assistants], but there may be cases where you 
+want to use one that is not currently supported.
+
+This tutorial walks you through the basics of creating an assistant for LLMs that are not
+currently supported. 
+"""
+
+# %%
+# ## The Finished Product
+
+# %%
+# !!! note
+#
+#     The code snippet below only includes up to Step 1. To actually include your source
+#     storage in Ragna, you must still perform Step 2.
+
+from ragna.core import Assistant
+
+
+class TutorialAssistant(Assistant):
+    """
+    This is a basic assistant created for didactic purposes
+    """
+
+    def answer(self, prompt, sources):
+        """Answer a prompt given some sources.
+
+        Args:
+            prompt: Prompt to be answered.
+            sources: Sources to use when answering answer the prompt.
+
+        Returns:
+            Answer.
+        """
```

Comment:
Maybe we shouldn't add a docstring here given that we cannot link to `Source` or the like. Maybe we should just explain the method and its inputs in a block before we show the implementation?